base/point.h

```diff
@@ -25,13 +25,19 @@ class Point {
    * \brief Standard destructor.
    */
   virtual ~Point();
-  /** General accessor method for the underlying data.
+
+  /** General accessor method for each feature of the underlying data.
    * \param feature_id The coordinate of interest. Indexing is implementation
    * specific.
    * \return The value of the coordinate specified in the input.
    */
   virtual double GetFeatureValue(unsigned feature_id) const = 0;
 
+  /** General accessor method for the label of the underlying data.
+   * \return The label of the object.
+   */
+  virtual double GetLabel() const = 0;
```

Comment:
But if its not 100% of the time, is it good to leave it here? I think it happens on the PredictProbabily function of the learner.h too, as it's more specific to a classification problem.
And about an id for the point, do you think its not necessary?